Finjector.Web/Handlers/IamIdClaimFallbackTransformer.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Security.Claims;
+using Microsoft.AspNetCore.Authentication;
+using Finjector.Web.Models;
+using Microsoft.Extensions.Options;
+using Ietws;
+
+namespace Finjector.Web.Handlers
+{
+    public class IamIdClaimFallbackTransformer : IClaimsTransformation
+    {
+        public const string ClaimType = "ucdPersonIAMID";
+        private readonly AuthOptions _authOptions;
+
+        public IamIdClaimFallbackTransformer(IOptions<AuthOptions> authOptions)
+        {
+            _authOptions = authOptions.Value;
+        }
+
+        public async Task<ClaimsPrincipal> TransformAsync(ClaimsPrincipal principal)
+        {
+            var iamId = principal.FindFirstValue(ClaimType);
+            if (!string.IsNullOrWhiteSpace(iamId))
+            {
+                return principal;
+            }
+
+            // if we don't have an IAM ID, try to get it from kerberos and save it in the claims
+            var kerbId = principal.FindFirstValue(ClaimTypes.NameIdentifier);
+
+            if (string.IsNullOrWhiteSpace(kerbId))
+            {
+                return principal;
```

Comment:
Throw an exception if the kerb is missing?

Reply:
The controller already returns a `NotAuthorized` if it fails to find an iamid claim. I figured that was good enough.